packages/connect-node/src/connect-protocol.ts

```diff
@@ -92,6 +97,19 @@ export function createConnectProtocol(
                 "Unsupported Media Type"
               );
             }
+
+            if (
+              shouldRequireHeader &&
+              (req.headers[connectProtocolVersionHeader] ?? "") !==
+                connectProtocolVersion
```

Comment:
You can safely do the following, which is a bit more readable:

```suggestion
              shouldRequireHeader &&
              req.headers[connectProtocolVersionHeader] !==
                connectProtocolVersion
```